pkg/storaged/utils.js

```diff
@@ -790,8 +790,32 @@
     return children;
 }
 
-export function get_fstab_config_with_client(client, block, also_child_config) {
-    let config = block.Configuration.find(c => c[0] == "fstab");
+export function get_fstab_config_with_client(client, block, also_child_config, subvol) {
+    function match(c) {
+        if (c[0] != "fstab")
+            return false;
+        if (subvol !== undefined) {
+            if (!c[1].opts)
+                return false;
+
+            const opts = decode_filename(c[1].opts.v).split(",");
+            if (opts.indexOf("subvolid=" + subvol.id) >= 0)
+                return true;
+            if (opts.indexOf("subvol=" + subvol.pathname) >= 0)
+                return true;
+
+            // btrfs mounted without subvol argument.
+            const btrfs_volume = client.blocks_fsys_btrfs[block.path];
+            const default_subvolid = client.uuids_btrfs_default_subvol[btrfs_volume.data.uuid];
+            if (default_subvolid === subvol.id && !opts.find(o => o.indexOf("subvol=") >= 0 || o.indexOf("subvolid=") >= 0))
+                return true;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20240111-145706-fc330020-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)